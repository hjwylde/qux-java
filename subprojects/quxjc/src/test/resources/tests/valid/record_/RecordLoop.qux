void f([{any x}] anys):
    for a in anys:
        if a.x == null:
            return

        print a

void main([str] args):
    f([{x: 0}, {x: 1}, {x: 'foo'}, {x: null}, {x: 2}])
    f([{x: null}, {x: 'bar'}, {x: {'2'}}])
    f([])
    f([{x: {{{}}}}, {x: []}, {x: [{[null]}]}, {x: null}, {x: {}}])

