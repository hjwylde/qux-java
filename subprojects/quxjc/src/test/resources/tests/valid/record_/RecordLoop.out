{x: 0}
{x: 1}
{x: 'foo'}
{x: {{{}}}}
{x: []}
{x: [{[null]}]}
